src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1225,8 +1225,12 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
             // Dev11 searches for the key file in the current directory and assembly output directory.
             // We always look to base directory and then examine the search paths.
-            keyFileSearchPaths.Add(baseDirectory);
-            if (baseDirectory != outputDirectory)
+            if (!string.IsNullOrWhiteSpace(baseDirectory))
```

Comment:
Use `IsNullOrEmpty` here. It is possible to have directory names that consist of only white space characters. Example is `"\u200A"`

Reply:
Any concerns about the usage of `IsNullOrWhiteSpace` for `additionalReferenceDirectories` from [before this change](https://github.com/dotnet/roslyn/blob/b8fa3a23b0bdc5b75d656c1b08462df77bd64328/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs#L1217)?

```cs
if (!string.IsNullOrWhiteSpace(additionalReferenceDirectories))
```

Reply:
That code is wrong too then. We should change it as well. Guessing the author wasn't aware of this really odd corner case.

Reply:
Do you want it to change in this PR?

I'll at least add a test which ensures that whitespace-only directory names are added to the keyFileSearchPaths.

Reply:
Since you're here and it's small lets go ahead and change it too.

Reply:
Changed the usages, however `Parse` will try to resolve the baseDirectory and therefore getting a useful test that operates on a whitespace baseDirectory is a bit of a challenge (might need to create a temp folder in the cwd instead of with Temp.CreateDirectory())?